include/boost/thread/v2/thread.hpp

```diff
@@ -69,7 +69,11 @@ namespace boost
       using namespace chrono;
       if (d > duration<Rep, Period>::zero())
       {
-        steady_clock::time_point c_timeout = steady_clock::now() + ceil<nanoseconds>(d);
+#ifdef BOOST_THREAD_HAS_CONDATTR_SET_CLOCK_MONOTONIC
```

Comment:
Why not use internal_clock here?

Reply:
Didn't think to, but that's a good idea.